src/scenic/core/requirements.py

```diff
@@ -258,4 +237,141 @@ def __str__(self):
         if self.name:
             return self.name
         else:
-            return f'"{self.ty.value}" on line {self.line}'
\ No newline at end of file
+            return f'"{self.ty.value}" on line {self.line}'
+
+## Builtin Requirements
+class SamplingRequirement(ABC):
+    """ A requirement to be checked to validate a sample.
+
+    Args:
+        optional: Whether or not this requirement must be
+            checked to validate the sample. Optional samples
+            can be checked, and if ``False`` imply that the
+            sample is invalid, but do not need to be checked
+            if all non-optional requirements are satisfied.
+    """
+    def __init__(self, optional):
+        self.optional=optional
+        self.active = True
+
+    @abstractmethod
+    def falsifiedByInner(self, sample):
+        """Returns False if the requirement is falsifed, True otherwise"""
+
+    def falsifiedBy(self, sample):
+        assert self.active
+        return self.falsifiedByInner(sample)
+
+    @property
+    @abstractmethod
+    def violationMsg(self):
+        """Message to be printed if the requirement is violated"""
+        pass
+
+class IntersectionRequirement(SamplingRequirement):
+    def __init__(self, objA, objB, optional=False):
+        super().__init__(optional=optional)
+        self.objA = objA
+        self.objB = objB
+
+    def falsifiedByInner(self, sample):
+        objA = sample[self.objA]
+        objB = sample[self.objB]
+        if objA.allowCollisions or objB.allowCollisions:
+            return False
+        return objA.intersects(objB)
+
+    @property
+    def violationMsg(self):
+        return f"Intersection violation: {self.objA} intersects {self.objB}"
+
+class BlanketCollisionRequirement(SamplingRequirement):
+    def __init__(self, objects, optional=True):
+        super().__init__(optional=optional)
+        self.objects = objects
+
+    def falsifiedByInner(self, sample):
+        objects = tuple(sample[obj] for obj in self.objects)
+        cm = trimesh.collision.CollisionManager()
+        for obj in objects:
+            if not obj.allowCollisions:
+                cm.add_object(str(obj), obj.occupiedSpace.mesh)
```

Comment:
The trimesh [docs](https://trimsh.org/trimesh.collision.html#trimesh.collision.CollisionManager.add_object) say this method will replace an existing object with the same name, which could easily happen (as the default name for an object is "unnamed Object"). Probably better to use `for i, obj in enumerate(objects)` above and then use `str(i)` as the name.

Reply:
Good catch!